impl/src/main/java/io/kroxylicious/testing/kafka/invm/KraftLogDirUtil.java

```diff
@@ -0,0 +1,89 @@
+ */
+package io.kroxylicious.testing.kafka.invm;
+
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.common.metadata.FeatureLevelRecord;
+import org.apache.kafka.common.metadata.UserScramCredentialRecord;
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.metadata.bootstrap.BootstrapMetadata;
+import org.apache.kafka.server.common.ApiMessageAndVersion;
+import org.apache.kafka.server.common.MetadataVersion;
+
+import kafka.server.KafkaConfig;
+import kafka.tools.StorageTool;
+import scala.collection.immutable.Seq;
+
+/**
+ * Note that this code is base on code from Kafka's StorageTool.
+ */
+final class KraftLogDirUtil {
+    private static final PrintStream LOGGING_PRINT_STREAM = LoggingPrintStream.loggingPrintStream(InVMKafkaCluster.LOGGER, System.Logger.Level.DEBUG);
+
+    private KraftLogDirUtil() {
+        throw new IllegalStateException();
+    }
+
+    static void prepareLogDirsForKraft(String clusterId, KafkaConfig config, List<UserScramCredentialRecord> scramCredentialRecords) {
+        var directories = StorageTool.configToLogDirectories(config);
+        try {
+            // Default the metadata version from the IBP version specified in config in the same way as kafka.tools.StorageTool.
+            var metadataVersion = MetadataVersion.fromVersionString(
+                    config.interBrokerProtocolVersionString() == null ? MetadataVersion.LATEST_PRODUCTION.version() : config.interBrokerProtocolVersionString());
+
+            // in kafka 3.7.0 the MetadataProperties class moved package, we use reflection to enable the extension to work with
+            // the old and new class.
+            var metaProperties = buildMetadataPropertiesReflectively(clusterId, config);
+            var bootstrapMetadata = buildBootstrapMetadata(scramCredentialRecords, metadataVersion);
+
+            formatReflectively(metaProperties, directories, bootstrapMetadata, metadataVersion);
+        }
+        catch (Exception e) {
+            throw new RuntimeException("failed to prepare log dirs for KRaft", e);
+        }
+    }
+
+    private static void formatReflectively(Object metaProperties, Seq<String> directories, BootstrapMetadata bootstrapMetadata, MetadataVersion metadataVersion)
+            throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+        var formatMethod = StorageTool.class.getDeclaredMethod("formatCommand", PrintStream.class, Seq.class, metaProperties.getClass(), BootstrapMetadata.class,
+                MetadataVersion.class,
+                boolean.class);
+        // note ignoreFormatter=true so tolerate a log directory which is already formatted. this is
+        // required to support start/stop.
+        formatMethod.invoke(null, LOGGING_PRINT_STREAM, directories, metaProperties, bootstrapMetadata, metadataVersion, true);
```

Comment:
nit: extract `ignoreFormatter=true` as a constant. e.g `IGNORE_FORMATTER` or maybe `IGNORE_FORMATTED` so the call is a bit more self describing.